lib/block_reservation.rb

```diff
@@ -0,0 +1,18 @@
+# block_reservation
+require 'date'
+
+module HotelManagment
+  class BlockReservation < Reservation
+
+    def initialize(first_name, last_name, check_in_date, check_out_date, room_number)
+      super
+      @cost = cost
+    end
+
+    def block_rate
```

Comment:
Shouldn't you simply override the `rate` method?

Why create the same method by a different name in a subclass?